easybuild/easyconfigs/m/MotionCor2/Motioncor2-wrapper.sh

```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+
+# Wrap around Motioncor2 to load the required toolchain so it can use the
+# expected version of CUDA.
+
+# Save path to the install tree.
+mc_base=$EBROOTMOTIONCOR2
+
+ml purge > /dev/null 2>&1
+ml CUDA/#CUDAVER#
```

Comment:
@akesandgren Oh boy, is this really necessary? Why not just use `CUDA` as a proper runtime dep, and include it in `versionsuffix`?

I may have other modules already that you shouldn't be purging, this is not an approach that'll work in general?

Reply:
The reason for this is that MotionCor2 (and Gctf) are dependencies for RELION (and Scipion) but they are built with fosscuda/2019a, which uses CUDA 10.1, and neither of them can use CUDA 10.1. So a wrapper the purges the modules and loads the specific CUDA and runs just the binary is basically the only way to do it. And it doesn't interfere with (in these two cases) with whatever other modules you have since it's only when running the motioncor/gctf binaries.

Reply:
OK, but running `module purge` in general is not a good idea, you may break the environment for some sites when doing that (it was an issue for us before, but not anymore now).

You should also make the wrapper more general, i.e. use full `module` commands, rather than using the `ml` shorthand (which may not be there when not using `Lmod`).

Reply:
In this case, doing purge is not a problem since the only thing done in the wrapper is to run the prepackaged binary and nothing else. I.e. the environment for it only depends on CUDA.

Does module purge exist for the other module environments? Can't remember any longer...

Reply:
@akesandgren Why not installing it with `fosscuda`? Even if it is binary distributed, it definitely solves the dependency problems.

Reply:
Because Motioncor2 is a binary distribution locked to specific CUDA versions.
And RELION (and Scipion) depends on it. But they will be built with the newest available xxxcuda toolchain. Thus MotionCor2 (and Gctf) has to do this little dance with a wrapper loading the CUDA version they can actually use.

I.e., using fosscuda to install it will be useless since it would be at the wrong version level.
The only way to have it as a dependency for RELION is to keep it at Core.